modules/terraform/cmd.go

```diff
@@ -17,6 +17,17 @@ func GetCommonOptions(options *Options, args ...string) (*Options, []string) {
 		args = append(args, "-no-color")
 	}
 
+	// If Lock is set to true, override the lock argument that has been set (ignore false)
+	if options.Lock == true {
+		lockKey := "-lock"
+		for i, argValue := range args {
```

Comment:
Oh... Instead of doing this complicated reverse-engineering, we should update those `apply`, `destroy`, etc commands to directly set the `-lock` parameter based on what it's set to in `terraform.Options`.